src/dbnode/persist/fs/retriever.go

```diff
@@ -299,13 +306,25 @@ func (r *blockRetriever) fetchBatch(
 
 	// Sort the requests by offset into the file before seeking
 	// to ensure all seeks are in ascending order
+	var limitErr error
 	for _, req := range reqs {
+		if limitErr != nil {
+			req.onError(limitErr)
+			continue
+		}
+
 		entry, err := seeker.SeekIndexEntry(req.id, seekerResources)
 		if err != nil && err != errSeekIDNotFound {
 			req.onError(err)
 			continue
 		}
 
+		if err := r.opts.QueryLimits().BytesReadLimit().Inc(int(entry.Size)); err != nil {
```

Comment:
I would probably cache the "bytesReadLimit()" call somewhere so don't need to do two method calls before the .Inc(...) in this tight loop.